eval.py

```diff
@@ -182,62 +163,60 @@ def main():
+    all_computed_metrics = {}
     for metric in metrics:
-        logging.info(f"Computing metric {metric}...")
-        try:
-            metric_evaluator = METRIC_NAME_TO_EVALUATOR[metric]["evaluator"]
-        except:
-            logging.error(f"Provided metric {metric} does not exist")
-            continue
-
         if metric == "aesthetic_predictor":
-            model_path = os.path.join(args.cache_dir, "aesthetic_predictor/model.pth")
-            download_model(url=args.aesthetic_predictor_model_url, output_path=model_path)
-            evaluator = AestheticPredictorEvaluator(device, model_path)
+            evaluator = AestheticPredictorEvaluator(device, args.aesthetic_predictor_model_url)
         elif metric == "human_preference_score":
-            model_path = os.path.join(args.cache_dir, "human_preference_score/model.pth")
-            download_model(url=args.human_preference_score_model_url, output_path=model_path)
-            evaluator = HumanPreferenceScoreEvaluator(device, model_path)
+            evaluator = HumanPreferenceScoreEvaluator(device, args.human_preference_score_model_url)
         else:
+            metric_evaluator = METRIC_NAME_TO_EVALUATOR[metric]["evaluator"]
             evaluator = metric_evaluator(device)
 
-        if isinstance(evaluator, BaseReferenceFreeEvaluator):
-            computed_metric = evaluator.evaluate(generated_images, prompts)
-        else:
-            assert isinstance(evaluator, BaseWithReferenceEvaluator)
-            computed_metric = evaluator.evaluate(generated_images, real_images)
-
-        for key, value in computed_metric.items():
-            if isinstance(value, torch.Tensor):
-                computed_metrics.append([key, value.item()])
-            elif isinstance(value, tuple):
-                computed_metrics.append([key, [metric.item() for metric in value]])
-            elif isinstance(value, float):
-                computed_metrics.append([key, value])
-            else:
-                raise RuntimeError(f"Unexpected type for computed metric: {type(computed_metric)}")
+        if not evaluator.is_useful(generated_images) and not metric in metrics_explicitly_specified:
+            logging.warning(f"Skipping metric {metric} as it is not useful for the given images.")
+            continue
+
+        logging.info(f"Computing metric {metric}...")
+        computed_metrics = evaluator.evaluate(generated_images,
+                                              real_images=real_images,
+                                              prompts=prompts)
+        all_computed_metrics.update(computed_metrics)
 
     # Print all results
-    print(tabulate(computed_metrics, headers=["Metric Name", "Value"], tablefmt="orgtbl"))
+    print(tabulate(all_computed_metrics.items(),
+                   headers=["Metric Name", "Value"],
+                   tablefmt="orgtbl"))
 
 
 if __name__ == "__main__":
-    main()
+    # https://github.com/huggingface/transformers/issues/26275
```

Comment:
can you add a comment here instead of just including a link without any context about what/why it's here